#ifndef TESTFORCLIENT_H
#define TESTFORCLIENT_H

#include <QDialog>
#include <QMessageBox>

namespace Ui {
class testForClient;
}

class testForClient : public QDialog
{
    Q_OBJECT

public:
    explicit testForClient(QWidget *parent = nullptr);
    ~testForClient();

private slots:
    void on_pushAnswer_clicked();

private:
    Ui::testForClient *ui;
    // std::vector<QString> rasstroystva = {"Депресивное растройство", "Тревожное растройство", "Растройство пищевого поведения", "Растройство адаптации", "Схизофрения",
    //                                    "Растройство внимания и гепиреактивности", "Растройство сна.", "Наркологическое растройство."};

    std::vector<QString> questions = {"Как долго вы уже замечаете симптомы депресии?",
                                      "Как часто у вас возникают панические приступы с сильным страхом \nи физиологическими симптомами (например, сердцебиение, потливость, дрожь)?",
                                      "Как часто вы занимаетесь необычными или отклоняющимися от нормы \nповеденческими практиками, связанными с пищей\n(например, ограничение калорийного потребления, \nпринятие слабительных, повышенная забота о пищевых продуктах)?",
                                      "Часто ли вы находитесь в стрессовом сотоянии?",
                                      "Как часто вы замечаете наличие следующих симптомов: расщепленность\nмышления (трудности в связности и логике мыслей),\nгаллюцинации (слышите голоса или видите нереальные вещи)\n и делириум (несвязное, бессмысленное речевое проявление)?",
                                      "Как часто у вас возникают эпизоды чрезмерной активности, повышенного\nнастроения и необычно быстрых мыслей?",
                                      "Как часто вы испытываете проблемы со сном, такие как бессонница \n(трудности со засыпанием или поддержанием сна), \nнарколепсия (неожиданное и непреодолимое ощущение сильной сонливости в течение дня), апноэ сна (прекращение дыхания во время сна)\n или синдром беспокойных ног (неприятные или неудержимые ощущения в ногах, вызывающие необходимость двигаться)?",
                                      "Часто ли вы испытываете стремление злоупотреблять алкоголем наркотиками \nили другими веществами?"};


    //std::vector<int> path_test();
    std::vector<int> answers_balance;
    int countQuestions = 8;
    int current_question;

signals:
    void VectorSaved(std::vector<int> data);

};

#endif // TESTFORCLIENT_H
